packages/widget/src/hooks/useCreateCosmosWallets.tsx

```diff
@@ -164,16 +156,17 @@ export const useCreateCosmosWallets = () => {
           },
           getAddress: async ({ signRequired }) => {
             try {
-              if (signRequired) {
-                throw new Error("always prompt wallet connection");
-              }
-              if (!chainId) {
-                throw new Error("no chain id");
+              const getAddressWithoutConnectingWallet =
+                cosmosWallet && currentWallet !== wallet && !signRequired && chainId;
+
+              if (getAddressWithoutConnectingWallet) {
+                const address = (await getWallet(wallet).getKey(chainId)).bech32Address;
+                return { address };
```

Comment:
@codingki is there a way to achieve this behavior for EVM/SVM ?

Reply:
for svm I believe it works, but we can't test this because there are no routes svm -> svm

Reply:
ah ok, i guess we don't need to worry about those for now 😆